tcwebhooks-core/src/main/java/webhook/teamcity/payload/content/WebHooksChange.java

```diff
@@ -5,15 +5,34 @@
 
 import jetbrains.buildServer.vcs.SVcsModification;
 import jetbrains.buildServer.vcs.VcsFileModification;
+import jetbrains.buildServer.vcs.VcsRootInstance;
+import org.jetbrains.annotations.Nullable;
 
 
 public class WebHooksChange {
 
+	private static @Nullable String tryGetVcsRootName(final SVcsModification modification) {
+		if(modification.isPersonal()) {
+			return null;
+		}
+
+		final VcsRootInstance vcsRoot;
+		try {
+			vcsRoot = modification.getVcsRoot();
```

Comment:
Can one modify a final in this instance?

Reply:
Oh, yes. It looks like it's possible because it's an initialisation, not a modification. Ignore this question.